src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -17,5 +17,27 @@ public abstract class AssemblyLoadContext
         public static event ResolveEventHandler TypeResolve;
         public static event ResolveEventHandler ResourceResolve;
         public static event ResolveEventHandler AssemblyResolve;
+
+        public event Func<AssemblyLoadContext, AssemblyName, Assembly> Resolving;
+        public event Action<AssemblyLoadContext> Unloading;
+
+        private static AssemblyLoadContext s_defaultContext = new DefaultAssemblyLoadContext();
```

Comment:
There is also AssemblyContext.Default property that you can implement while you are on it, so this would become:

`public static AssemblyLoadContext Default { get; } = new DefaultAssemblyLoadContext();`